db/migrate/20201015145100_add_progress_field_on_trainees.rb

```diff
@@ -0,0 +1,9 @@
+class AddProgressFieldOnTrainees < ActiveRecord::Migration[6.0]
+  def change
+    change_table :trainees, bulk: true do |t|
+      t.jsonb :progress, default: {}
+    end
+
+    add_index :trainees, :progress, using: :gin
```

Comment:
just wondering as I need a drink ...


what is `gin` and in relation to  jsonb.

Reply:
It's a type of index for jsonb which is a bit more efficient for traversing a large json structure. Might be unnecessary if we're only going to hold a few keys under this progress

Reply:
do we need something like 

```sql
CREATE EXTENSION IF NOT EXISTS pg_trgm;
```